app/assets/javascripts/analytics/pii.js

```diff
@@ -0,0 +1,64 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+  var EMAIL_PATTERN = /[^\s=/?&]+(?:@|%40)[^\s=/?&]+/g
+  var POSTCODE_PATTERN = /[A-PR-UWYZ][A-HJ-Z]?[0-9][0-9A-HJKMNPR-Y]?(?:[\s+]|%20)*[0-9][ABD-HJLNPQ-Z]{2}/gi
+  var DATE_PATTERN = /\d{4}(-?)\d{2}(-?)\d{2}/g
+
+  var pii = function () {
+    this.stripDatePII = false
+    this.stripPostcodePII = false
+  }
+
+  pii.prototype.stripPII = function (value) {
+    if (typeof value === 'string') {
+      return this.stripPIIFromString(value)
+    } else if (Object.prototype.toString.call(value) === '[object Array]' || Object.prototype.toString.call(value) === '[object Arguments]') {
+      return this.stripPIIFromArray(value)
+    } else if (typeof value === 'object') {
+      return this.stripPIIFromObject(value)
+    } else {
+      return value
```

Comment:
A number could be PII but it would be slightly obfuscated (e.g. 20190911, today's date) which would make it both hard to identify as PII and hard to strip out. 

I don't know why the code only acts on these three variable types, presumably it only gets passed those three.